app_widgetbook/lib/core/ui_elements/app_elevated_button.dart

```diff
@@ -0,0 +1,167 @@
+import 'package:app_widgetbook/utils.dart';
+import 'package:core/core.dart';
+import 'package:flutter/material.dart';
+import 'package:widgetbook/widgetbook.dart' show Knobs, Option;
+import 'package:widgetbook_annotation/widgetbook_annotation.dart';
+
+/// Use case for a Primary Button style
+@WidgetbookUseCase(name: 'Custom', type: AppElevatedButton)
+Widget customAppElevatedButtonUseCase(BuildContext context) {
+  return AppElevatedButton(
+    label: context.knobs.nullableText(
+      label: 'Label',
+      initialValue: 'Button',
+    ),
+    icon: context.knobs.boolean(label: 'Has Icon', initialValue: true)
+        ? context.knobs.options(
+            label: 'Icon',
+            options: buttonIconOptions,
+          )
+        : null,
+    onPressed: () {},
+    borderColor: context.knobs.options<Color?>(
+      label: 'Border Color',
+      options: const [
+        Option(
+          label: 'None',
+          value: null,
+        ),
+        Option(
+          label: 'Primary',
+          value: AppColors.primary,
+        ),
+        Option(
+          label: 'Secondary',
+          value: AppColors.secondary,
+        ),
+        Option(
+          label: 'Pink',
+          value: AppColors.pink,
+        ),
+        Option(
+          label: 'White',
+          value: AppColors.white,
+        ),
+      ],
+    ),
+    backgroundColor: context.knobs.options<Color?>(
+      label: 'Background Color',
+      options: const [
+        Option(
+          label: 'Primary',
+          value: AppColors.primary,
+        ),
+        Option(
+          label: 'Secondary',
+          value: AppColors.secondary,
+        ),
+        Option(
+          label: 'Pink',
+          value: AppColors.pink,
+        ),
+        Option(
+          label: 'White',
+          value: AppColors.white,
+        ),
+      ],
+    ),
+    textColor: context.knobs.options<Color?>(
+      label: 'Text Color',
+      options: const [
+        Option(
+          label: 'White',
+          value: AppColors.white,
+        ),
+        Option(
+          label: 'Primary',
+          value: AppColors.primary,
+        ),
+        Option(
+          label: 'Secondary',
+          value: AppColors.secondary,
+        ),
+        Option(
+          label: 'Pink',
+          value: AppColors.pink,
+        ),
+      ],
+    ),
+    height: context.knobs.boolean(label: 'Is Large') ? 56 : 40,
```

Comment:
@jenshor I realize that it is not optimal to have non-reusable constants like this (the `56` and `40` values).
What I usually do is similar to the `AppSpacing` class I've mentioned for the grid gaps and padding/margin.
I can create another `AppSizing` class and include in it constants like `buttonHeight`, `buttonHeightLg`, `avatarSize`, `avatarSizeSm`, and so on. 
What do you think of this approach? I can work on it after adding the `AppSpacing` in a separate PR.

Reply:
Sounds good